### ISSUE TITLE
Error in fetch kafka.consumer.ConsumerFetcherThread$FetchRequest@552d6b60. Possible cause: java.nio.BufferUnderflowException

### ISSUE THREAD

**Issue by shawnbutts** [yahoo/kafka-manager#228]
Running 1.3.0.7 with Kafka 2.11-0.8.2.1.  Getting 100s of these errors in the logs.

[warn] k.c.ConsumerFetcherThread - [ConsumerFetcherThread-KafkaManagerOffsetCache_mon01-1461164220016-3d2546ed-0-3], Error in fetch kafka.consumer.ConsumerFetcherThread$FetchRequest@552d6b60. Possible cause: java.nio.BufferUnderflowException

**Comment by xavierpayne** [comment]
Running 1.3.0.8 with confluent kafka 9.0.1 and seeing the same.

**Comment by xiongjiaji** [comment]
Running 1.3.0.7 with Kafka 2.11-0.8.2.1. Getting the same warning log.

**Comment by jmwong** [comment]
Same here. On 1.3.0.8 with Kafka 2.11-0.8.2.2.

**Comment by aaronmray** [comment]
I'm pretty sure this has to do with the fact that kafka-manager is built against kafka 0.9.0.1. The consumer in 0.9.0.1 is not compatible with an 0.8.2.x broker. The reverse (0.8.2.x consumer with 0.9.0.1 broker) is backwards compatible if inter.broker.protocol.version is set correctly in the broker config. 

I don't know of an eloquent way to solve this dilemma other than trying to use 0.8.2.x dependencies and fixing any compilation errors. Alternatively, you could update your brokers to 0.9.0.1. 

I suppose a third option could be to abstract the consumer portion of kafka-manager out into its own microservice (or number of micoservices for each version of kafka you need to monitor). Then you could determine which microservice to call based on the version setting in the cluster configuration.

**Comment by wswuai** [comment]
Same here. On 1.3.0.8 with Kafka 2.11-0.8.2.2.

**Comment by DingHe** [comment]
Same here. On 1.3.0.8 with Kafka 2.11-0.8.2.2.

**Comment by jmwong** [comment]
I worked around this by creating a branch off 3ef1568cee03359387f786e1e8aec4f299ff794c. I applied these patches with some modifications to rip out the Kafka 0.9 stuff:
https://github.com/yahoo/kafka-manager/commit/fedbdca0a2a680f51bea65506a6b73a7125095b2
https://github.com/yahoo/kafka-manager/commit/0ab8108b0702df9d39c6695abdff97ce54be8a1f

The branch is here:
https://github.com/jmwong/kafka-manager/tree/pre-090-support

**Comment by Songdantes** [comment]
Same here. On 1.3.0.8 with Kafka 2.11-0.8.2.2.
![image](https://cloud.githubusercontent.com/assets/12936628/21561677/d317232c-ceaa-11e6-814f-9146bb424f90.png)
![image](https://cloud.githubusercontent.com/assets/12936628/21561706/354f3444-ceab-11e6-832c-5b73f3f9891b.png)
I don't know if this is the reason why  my JMX Metrics does‘t  display.

**Comment by xavierpayne** [comment]
Nope. JMX metrics usually don't display because of firewall rules. The tricky part is you need to whitelist everything. Not just the port to told JMX to listen on. This is because during the connection JMX creates random ports. :(
